### PR TITLE
feat: new ui for response interface generation

### PR DESCRIPTION
**Pull request by anwarulislam**
Closes HFE-533

### Description
Response interface generation, a new UI, has been added to the sidebar. It will be open in a slider from right to left when you click on `Generate data schema` in the `More` option. 

### Checks

- [x] My pull request adheres to the code style of this project
- [ ] My code requires changes to the documentation
- [ ] I have updated the documentation as required
- [x] All the tests have passed

### CODE REVIEW

**Review by nivedin — CHANGES_REQUESTED** [review]
- can we hide the response interface type section if the response is null

image
- the interface section layout height changes each time, also the component crashes sometime

https://github.com/hoppscotch/hoppscotch/assets/53208152/50b07f01-2079-4bd3-94f1-a477be2b704e

**Comment by liyasthomas** [comment]
I see some caveats for this design hence I'll review this and give comments internally over Slack.

File: packages/hoppscotch-common/src/components/app/PaneLayout.vue [hunk 1]
```diff
@@ -16,7 +16,7 @@
     >
       <Splitpanes
         class="smart-splitter"
-        :horizontal="COLUMN_LAYOUT"
+        :horizontal="COLUMN_LAYOUT || forceColumnLayout"
```
[the remark this answers is not in the document]
Comment (anwarulislam) [reply packages/hoppscotch-common/src/components/app/PaneLayout.vue:19]: I experimented with different implementations in the past, but now I don't think it's necessary. However, I believe it's good to have for the future.

**Review by nivedin — APPROVED** [review]
LGTM 💯

**Pull request merged**